src/DotLiquid.Tests/ContextTests.cs

```diff
@@ -278,6 +278,17 @@ public void TestVariableNotFoundExceptionIgnoredForUnlessStatement()
             Assert.AreEqual(0, template.Errors.Count);
         }
 
+        [Test]
+        public void TestVariableNotFoundExceptionFromInvalidVariableName()
+        {
+            Template template = Template.Parse("{{ . }}");
+            string rendered = template.Render();
+
+            Assert.AreEqual("", rendered);
+            Assert.AreEqual(1, template.Errors.Count);
+            Assert.AreEqual(string.Format(Liquid.ResourceManager.GetString("VariableNotFoundException"), "."), template.Errors[0].Message);

```

Comment:
Might want to follow the model of checking for exception similar to TestHashWithDefaultProc. This is checking a string output which is good but lacking type checking.

Reply:
I actually started looking into this - I see this has a different behavior and your test is valid.